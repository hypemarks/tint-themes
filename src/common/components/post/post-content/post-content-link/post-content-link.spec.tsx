import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import PostContentLink from './post-content-link';

const onClick = jest.fn();

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Test <PostContentLink /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<PostContentLink isButton={true} onClick={onClick} href='somesortofstring' />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render div with role button then fire onClick', () => {
    const wrapper = render(<PostContentLink isButton={true} onClick={onClick} href='somesortofstring' />);
    const postContentLink = wrapper.getByTestId('post-content-link-button');
    const postContentLinkByRole = wrapper.getByRole('button');
    fireEvent.click(postContentLink);
    expect(postContentLinkByRole).toBeInTheDocument();
    expect(onClick).toBeCalled();
  });

  it('should render anchor tag with specific href', () => {
    const wrapper = render(<PostContentLink isButton={false} onClick={onClick} href='somesortofstring' />);
    const postContentLinkAnchor = wrapper.getByTestId('post-content-link-anchor');
    expect(postContentLinkAnchor).toBeInTheDocument();
    expect(postContentLinkAnchor).toHaveAttribute('href', 'somesortofstring');
  });

  it('should render anchor tag with specific href and not fire onClick', () => {
    const wrapper = render(<PostContentLink isButton={false} onClick={onClick} href='somesortofstring' />);
    const postContentLinkAnchor = wrapper.getByTestId('post-content-link-anchor');
    fireEvent.click(postContentLinkAnchor);
    expect(onClick).not.toBeCalled();
  });
});
