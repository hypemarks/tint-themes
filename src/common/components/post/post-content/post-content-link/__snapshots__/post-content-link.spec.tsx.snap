// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostContentLink /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-content-link"
    data-testid="post-content-link-button"
    role="button"
  />
</DocumentFragment>
`;
